src/xenia/kernel/xam/xam_info.cc

```diff
@@ -123,6 +123,8 @@ dword_result_t keXamBuildResourceLocator(uint64_t module,
   }
   auto copy_count = std::min(size_t(buffer_count), path.size());
   xe::copy_and_swap(buffer_ptr.as<char16_t*>(), path.c_str(), copy_count);
+  (buffer_ptr.as<char16_t*>())[copy_count] = 0;
```

Comment:
I've checked once more.

No it doesn't include null terminator. However it is still incorrect. If copy_count matches or exceed buffer_count null terminator should be omitted.

More here: http://docwiki.embarcadero.com/RADStudio/Sydney/en/Snprintf;snwprintf

Reply:
`path.c_str()` will always be null-terminated since it's `c_str` of a variant of `std::string`, since for C strings, it has to be present to indicate termination. I was talking about the calling contract of `keXamBuildResourceLocator` (not `snprintf`), and the target string, `buffer_ptr[buffer_count]`, rather than the temporary `path`. I'd expect NT kernel, Windows API, and other Microsoft system functions to always return null-terminated strings, for safety, even if they had to be truncated because the buffer provided by the user is too small to fit the entire string. C standard library functions are very inconsistent, but Microsoft's functions, like GetFullPathName, GetWindowText (with some exceptions like GetEnvironmentVariable, where the written data is claimed to be undefined if the buffer is too small) usually accept the wanted _buffer size_ rather than _string length (character count)_ — for a _buffer including the text **and** the null terminator_. And if the buffer is too small, for the entire string, `buffer_count - 1` characters are written, and the null terminator is written at `buffer_ptr[std::min(copy_count, buffer_count - 1)]`.